scm-record/src/types.rs

```diff
@@ -74,73 +74,81 @@ pub enum RecordError {
     Bug(String),
 }
 
-/// The Unix file mode. The special mode `0` indicates that the file did not exist.
+/// The file mode.
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct FileMode(pub usize);
+pub enum FileMode {
+    /// A Read Write Execute style Unix file mode
+    Rwx(usize),
+
+    /// A special file mode indicating that the file was not present.
+    Absent,
+}
 
 impl FileMode {
-    /// Get the file mode corresponding to an absent file. This typically
-    /// indicates that the file was created (if the "before" mode) or deleted
-    /// (if the "after" mode).
-    pub fn absent() -> Self {
-        Self(0)
-    }
+    /// The default Unix permissions for files.
+    pub const FILE_DEFAULT: FileMode = FileMode::Rwx(0o100644);
```

Comment:
I've added this so that the tests that previously passed `None` don't each have to dupe `100644`.